### PR TITLE
feat(CI): add support for GCC

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
Add support for GCC:

- ubuntu 20.04 with gcc-10
- ubuntu 20.04 with gcc-9 (default)
- ubuntu 18.04 with gcc-7 (default)

Also:
- remove clang9 with modules, since checking modules with clang 10 is enough
- remove clang6 with ubuntu 20.04, as we already test it in ubuntu 18.04
- rename `clang10` to just `clang` in 20.04, as it's the default one (consistency)

Co-authored-by Viste <viste02@gmail.com>

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
closing in favour of: https://github.com/azerothcore/azerothcore-wotlk/pull/3315

File: src/server/game/Battlegrounds/Zones/BattlegroundAV.cpp [hunk 1]
```diff
@@ -196,7 +196,12 @@ void BattlegroundAV::HandleQuestComplete(uint32 questid, Player* player)
             break;
         case AV_QUEST_A_BOSS1:
         case AV_QUEST_H_BOSS1:
-            m_Team_QuestStatus[teamId][4] += 9; //you can turn in 10 or 1 item..
+            m_Team_QuestStatus[teamId][4] += 10; //you can turn in 10 or 1 item..
+#if defined(ENABLE_EXTRAS) && defined(ENABLE_EXTRA_LOGS)
+            if (m_Team_QuestStatus[teamId][4] >= 200)
+                sLog->outDebug(LOG_FILTER_BATTLEGROUND, "BG_AV Quest %i completed (need to implement some events here", questid);
```
Comment (Helias) [inline src/server/game/Battlegrounds/Zones/BattlegroundAV.cpp:202]:
```suggestion
                sLog->outDebug(LOG_FILTER_BATTLEGROUND, "BG_AV Quest %i completed (need to implement some events here)", questid);
```

File: src/server/game/Spells/Auras/SpellAuras.cpp [hunk 2]
```diff
@@ -2636,6 +2637,7 @@ void UnitAura::FillTargetMap(std::map<Unit*, uint8> & targets, Unit* caster)
                         break;
                     }
                     case SPELL_EFFECT_APPLY_AREA_AURA_PET:
+                        #pragma GCC diagnostic ignored "-Wimplicit-fallthrough"
```
Comment (Winfidonarleyan) [inline src/server/game/Spells/Auras/SpellAuras.cpp:2640]: wow wow wow, stop, please and loot this https://en.cppreference.com/w/cpp/language/attributes/fallthrough

Comment (FrancescoBorzi) [reply src/server/game/Spells/Auras/SpellAuras.cpp:2640]: @Winfidonarleyan yes I will add this for sure :)

Comment (Winfidonarleyan) [reply src/server/game/Spells/Auras/SpellAuras.cpp:2640]: Okay

File: deps/SFMT/SFMT.h [hunk 3]
```diff
@@ -168,6 +168,9 @@ class SFMTRand {                              // Encapsulate random number gener
         uint32_t y = seed;                  // Temporary
         uint32_t statesize = SFMT_N*4;      // Size of state vector
 
+// TODO: manually silencing this warning, but it should properly fixed
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
```
Comment (Winfidonarleyan) [inline deps/SFMT/SFMT.h:173]: https://github.com/TrinityCore/TrinityCore/commit/dd55e41c8158fe1f60e120b0b89469d4112158d2#diff-9d83efc6463a522a391b759b3cf9e769

Comment (FrancescoBorzi) [reply deps/SFMT/SFMT.h:173]: thanks @Winfidonarleyan

File: src/server/scripts/Kalimdor/ZulFarrak/instance_zulfarrak.cpp [hunk 4]
```diff
@@ -139,7 +139,7 @@ class spell_zulfarrak_unlocking : public SpellScriptLoader
                 std::list<WorldObject*> cagesList;
                 acore::AllWorldObjectsInRange objects(GetCaster(), 15.0f);
                 acore::WorldObjectListSearcher<acore::AllWorldObjectsInRange> searcher(GetCaster(), cagesList, objects);
-                GetCaster()->VisitNearbyObject(15.0f, searcher);
+//                GetCaster()->VisitNearbyObject(15.0f, searcher);
```
Comment (FrancescoBorzi) [inline src/server/scripts/Kalimdor/ZulFarrak/instance_zulfarrak.cpp:142]: @azerothcore/developers @Winfidonarleyan @Viste @Rochet2 @Shard-MW @vhiperdev @Nefertumm @Kaev  I need help with these 2 lines. If we uncomment them, they will cause GCC non-PCH build to fail.

Any ideas?

Comment (Nefertumm) [reply src/server/scripts/Kalimdor/ZulFarrak/instance_zulfarrak.cpp:142]: Is there an error code or something?

Comment (FrancescoBorzi) [reply src/server/scripts/Kalimdor/ZulFarrak/instance_zulfarrak.cpp:142]: @Nefertumm 

image

Comment (Viste) [reply src/server/scripts/Kalimdor/ZulFarrak/instance_zulfarrak.cpp:142]: hmm i dont have this on GCC 10.2.0

Comment (Viste) [reply src/server/scripts/Kalimdor/ZulFarrak/instance_zulfarrak.cpp:142]: you need add 
```
#include "GridNotifiers.h"
#include "GridNotifiersImpl.h"```
to instance_zulfarrak.cpp 
and remove #include `"GridNotifiers.h"` from zulfarrak.h

**Review by Helias — APPROVED** [review]

**Pull request merged**